sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -735,7 +735,7 @@ colTypeList
     ;
 
 colType
-    : identifier dataType (COMMENT STRING)?
+    : identifier dataType (COMMENT STRING)? (DEFAULT defaultExpression=expression)?
```

Comment:
Is that `defaultExpression=expression` scope too big for DDL default constraint? In my memory, the common default constraint are `NULL`, `NUMBER`, `STRING`, `CURRENT_DATE`, `CURRENT_TIMESTAMP`.

Reply:
> Is that `defaultExpression=expression` scope too big for DDL default constraint? In my memory, the common default constraint are `NULL`, `NUMBER`, `STRING`, `CURRENT_DATE`, `CURRENT_TIMESTAMP`.

Thanks for your review. As your said, The description of this PR contains a discussion about the scope of default constraint. Do we need to implement other expressions, like `Cast(1 as float)`, `1 + 2` and so on ?

Reply:
This is Oracle's default constraint. https://docs.oracle.com/javadb/10.8.3.0/ref/rrefsqlj30540.html#rrefsqlj30540__sqlj64478

Reply:
@lipzhu It's worth to reference, but we need to look at the actual situation on Spark SQL.

Reply:
@lipzhu I reduced the scope of default constraint.  Thanks.